xbmc/platform/linux/input/LibInputHandler.cpp

```diff
@@ -63,6 +64,40 @@ static void LogHandler(libinput  __attribute__((unused)) *libinput, libinput_log
   }
 }
 
+bool CLibInputHandler::CheckInputGroup()
+{
+  auto numGroups = getgroups(0, nullptr);
+  if (numGroups < 0)
+  {
+    CLog::Log(LOGERROR, "CLibInputHandler::{} - unable to retrieve groups: {}", __FUNCTION__, strerror(errno));
+    return false;
+  }
+
+  std::vector<gid_t> groups(numGroups);
+
+  auto ret = getgroups(numGroups, groups.data());
+  if (ret < 0)
+  {
+    CLog::Log(LOGERROR, "CLibInputHandler::{} - unable to retrieve groups: {}", __FUNCTION__, strerror(errno));
+    return false;
+  }
+
+  auto groupId = getgrnam("input");
+  if (groupId)
+  {
+    for (const auto& group : groups)
```

Comment:
using algorithm from the STL is preferred

```suggestion
    return std::find_if(groups.cbegin(), groups.cend() [&](const gid_t group){ return group == groupId->gr_gid; }) != groups.cend();
```